### PR TITLE
fix non-functional cancel button for broken primaries

### PR DESCRIPTION
**Pull request by lloyd**

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Link to #2839

**Comment by shane-tomlinson** [comment]
Does this fix a non-functional cancel button for broken primaries or just avoid it all together by going to the "primary not available" screen?

File: resources/static/dialog/js/modules/pick_email.js [hunk 1]
```diff
@@ -61,8 +61,10 @@ BrowserID.Modules.PickEmail = (function() {
     var record = checkEmail.call(self, email);
     if (!! record) {
       dialogHelpers.refreshEmailInfo.call(self, email, function (info) {
-	record = checkEmail.call(self, email);
-        if (record.cert)
+        record = checkEmail.call(self, email);
+        if ("offline" === info.state)
```
Comment (shane-tomlinson) [inline resources/static/dialog/js/modules/pick_email.js]: Going along with https://github.com/mozilla/browserid/pull/2843#discussion_r2358831, this should have a test as well as have some {}s.

**Comment by lloyd** [comment]
Avoids it all together by fixing screen transitions.

-- lloyd (thumb-typing)

File: resources/static/dialog/js/modules/pick_email.js [hunk 2]
```diff
@@ -61,8 +61,10 @@ BrowserID.Modules.PickEmail = (function() {
     var record = checkEmail.call(self, email);
     if (!! record) {
```
Comment (shane-tomlinson) [inline resources/static/dialog/js/modules/pick_email.js:62]: Does the record need to be checked for here? No action is taken if a record is not present.

**Comment by shane-tomlinson** [comment]
@lloyd - I rebased this off of branch issue3834 and added a new test. Mind reviewing?

**Comment by lloyd** [comment]
r+ with shane's improvments

**Pull request merged**